### PR TITLE
Update rangeSelector to include month/year for time

### PR DESCRIPTION
**Pull request by mattrout92**
### What

- Refactored the range selector for time to include month/year selection
- Fixed minor bug on list selection (was using the wrong id when adding to filter api)

### How to review

- Start renderer and filter-dataset-controller on feature/refactor-start-end
- Run the rest of cmd services on cmd-develop
- Verify that when you choose the time dimension that you can select month/year from drop down menu successfully

### Who can review
Anyone

### CODE REVIEW

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 1]
```diff
@@ -25,7 +25,7 @@
                         </div>
                      </details>
                   </div>
-                  <form action="{{$.Data.RangeData.URL}}" method="post">
+                  <form action="{{$.Data.RangeData.URL}}" method="post">
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: Remove whitespace at end of line.

File: vendor/github.com/ONSdigital/dp-frontend-models/model/dataset-filter/rangeSelector/range-selector.go [hunk 2]
```diff
@@ -11,20 +11,21 @@ type Page struct {
 
 // RangeSelector ...
```
Comment (nshumoogum) [inline vendor/github.com/ONSdigital/dp-frontend-models/model/dataset-filter/rangeSelector/range-selector.go]: Exported struct needs comment

File: vendor/github.com/ONSdigital/dp-frontend-models/model/dataset-filter/rangeSelector/range-selector.go [hunk 3]
```diff
@@ -39,6 +40,13 @@ type Filter struct {
...
 }
 
+// DateRange ...
+type DateRange struct {
+	Range
+	MonthValues []string `json:"month_values"`
+	YearValues  []string `json:"year_values"`
+}
+
 // Range ...
```
Comment (nshumoogum) [inline vendor/github.com/ONSdigital/dp-frontend-models/model/dataset-filter/rangeSelector/range-selector.go]: Several structs in file that are exported with no comment.

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 4]
```diff
@@ -35,6 +35,67 @@
                         		<input class="btn btn--link underline-link" type="submit" value="Remove all" name="remove-all-range" />
 							  </div>
                               <div class="clearfix">
+                                 {{if eq $.Data.Title "Time"}}
+                                 <div class="margin-bottom--1">
+                                    <span class="font-size-17"><strong>Start Date</strong></span>
+                                 </div>
+                                 <div class="col col--md-12 col--lg-12 margin-bottom--2">
+                                    <label class="block margin-bottom--1" for="Month">
+                                    <strong>Month</strong>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: I would indent this, within label tags.

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 5]
```diff
@@ -35,6 +35,67 @@
...
+                                    </label>
+                                    <div class="select-alt">
+                                       <select class="select width-sm--10 width-md--10 width-lg--10" name="start-month" id="start-month">
+                                          {{ range $.Data.DateRangeData.MonthValues }}
+                                          <option value="{{.}}">{{.}}</option>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: Again i would indent here as you are looping over month values; easier to recognise loop.

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 6]
```diff
@@ -35,6 +35,67 @@
...
+                                          {{ end }}
+                                       </select>
+                                    </div>
+                                 </div>
+                                 <div class="col col--lg-11 col--md-11 margin-bottom--2">
+                                    <label class="block margin-bottom--1" for="Year">
+                                    <strong>Year</strong>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: Indent as above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 7]
```diff
@@ -35,6 +35,67 @@
...
+                                    </label>
+                                    <div class="select-alt">
+                                       <select class="select width-sm--10 width-md--10 width-lg--10" name="start-year" id="start-year">
+                                          {{ range $.Data.DateRangeData.YearValues }}
+                                          <option value="{{.}}">{{.}}</option>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: Indent as above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 8]
```diff
@@ -35,6 +35,67 @@
...
+                                    </div>
+                                 </div>
+                              </div>
+                              <div class="clearfix">
+                                 <div class="margin-bottom--1">
+                                    <span class="font-size-17"><strong>End Date</strong></span>
+                                 </div>
+                                 <div class="col col--md-12 col--lg-12 margin-bottom--2">
+                                    <label class="block margin-bottom--1" for="Month">
+                                    <strong>Month</strong>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: As above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 9]
```diff
@@ -35,6 +35,67 @@
...
+                                    </label>
+                                    <div class="select-alt">
+                                       <select class="select width-sm--10 width-md--10 width-lg--10" name="end-month" id="end-month">
+                                          {{ range $.Data.DateRangeData.MonthValues }}
+                                          <option value="{{.}}">{{.}}</option>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: As above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 10]
```diff
@@ -35,6 +35,67 @@
...
+                                          {{ end }}
+                                       </select>
+                                    </div>
+                                 </div>
+                                 <div class="col col--lg-11 col--md-11 margin-bottom--2">
+                                    <label class="block margin-bottom--1" for="Year">
+                                    <strong>Year</strong>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: As above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 11]
```diff
@@ -35,6 +35,67 @@
...
+                                    </label>
+                                    <div class="select-alt">
+                                       <select class="select width-sm--10 width-md--10 width-lg--10" name="end-year" id="end-year">
+                                          {{ range $.Data.DateRangeData.YearValues }}
+                                          <option value="{{.}}">{{.}}</option>
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: As above

File: assets/templates/dataset-filter/range-selector.tmpl [hunk 12]
```diff
@@ -35,6 +35,67 @@
...
+                                          {{ end }}
+                                       </select>
+                                    </div>
+                                 </div>
+                                <div class="col col--lg-6 col--md-6 margin-top-md--3 margin-top-lg--3 text-right--lg text-right--md block">
+                                    <input type="submit" class="btn btn--link underline-link padding-top--1" value="Add range">
+                                </div>
+                                 {{else}}
```
Comment (nshumoogum) [inline assets/templates/dataset-filter/range-selector.tmpl]: Incorrect indentation from line 98 to 128; used 1 too many whitespaces for indentation.

**Comment by mattrout92** [comment]
I have indented the html tags which follow other html tags, but I don't feel as though indenting the tags which follow go-template code is necessary, as this will disappear when compiled anyway

**Pull request merged**